testng-core/src/test/java/test/listeners/github1130/GitHub1130Test.java

```diff
@@ -22,19 +21,20 @@ public void classListenerShouldBeOnlyInstantiatedOnce() {
   }
 
   private void checkGithub1130(TestNG tng) {
-    MyListener.count = 0;
-    MyListener.beforeSuiteCount = new ArrayList<>();
-    MyListener.beforeClassCount = new ArrayList<>();
     TestListenerAdapter adapter = new TestListenerAdapter();
     tng.addListener((ITestNGListener) adapter);
     tng.run();
     assertThat(adapter.getFailedTests()).isEmpty();
     assertThat(adapter.getSkippedTests()).isEmpty();
-    assertThat(MyListener.beforeSuiteCount.size()).isEqualTo(1);
-    assertThat(MyListener.beforeClassCount.size()).isEqualTo(2);
-    assertThat(MyListener.beforeSuiteCount.get(0))
-        .isEqualTo(MyListener.beforeClassCount.get(0))
-        .isEqualTo(MyListener.beforeClassCount.get(1));
-    assertThat(MyListener.count).isEqualTo(1);
+    // We cannot prevent TestNG from instantiating a listener multiple times
```

Comment:
It highlights a design issue in the listener lifecycle we should fix.
What is the supposed lifecycle and how users can manage states in listeners?
Maybe we can start by a specification and test cases.

Reply:
Yes I know.That's why I intentionally added this comment there so that this stands out. I think we should basically create a different issue that tracks all the discrepancies and fix it as part of that. It would be a lot more easier and focussed since we are looking at just one single aspect. Your thoughts ?

Reply:
Agree to create an issue in order to track it.

Reply:
Filed an issue to track this https://github.com/cbeust/testng/issues/2843